### ISSUE TITLE
posix_spawnp() should return int, not long

### ISSUE THREAD

**Issue by eregon** [jnr/jnr-posix#89]
posix_spawnp() should return an int and not a long.
https://github.com/jnr/jnr-posix/blob/master/src/main/java/jnr/posix/POSIX.java#L128-L133

This is as specified by the man page:
```
int posix_spawn(pid_t *pid, const char *path,
                       const posix_spawn_file_actions_t *file_actions,
                       const posix_spawnattr_t *attrp,
                       char *const argv[], char *const envp[]);
```
Of course, one problem is that a native int might 4 bytes, 8 bytes or something different than the Java 4-bytes int. But many other functions have this problem, so I  suppose primitive types take the native meaning and not necessarily Java's.

This means on Solaris, the returned value is actually wrong, and the real value is `(returnValue >> 32)`.
cc @nirvdrum @sanzinger

**Comment by eregon** [comment]
NOTE: Wayne changed it to long in 922259d97e116e6e6c4be3c248237c81939b17f9.

**Comment by eregon** [comment]
Actually, the jnr-posix version of this function returns a `long` to represent the output argument `pid` (and not the return value of the native call), but sometimes that value is not the correct `pid_t` (instead it's the `pid << 32`).
Maybe the proper thing would be to have posix_spawn (and posix_spawnp of course) return a `pid_t` in Java?
Otherwise fixing the conversion from `pid_t` to `long` would already return a sensible value.

**Comment by headius** [comment]
@eregon Yes, that seems sensible. We will probably need to add it as a different function name on POSIX interface.
